staff/puppet/gen-rootpw

```diff
@@ -1,28 +1,52 @@
-#!/bin/bash -e
-# Update the OCF root password. We use a ridiculous number of hash iterations
-# to make cracking the hashes way over-the-top impossible.
-
-ROOT_PW_FILE="/opt/puppetlabs/shares/private/rootpw"
-
-echo "########################################################"
-echo "This will update the root password for ALL OCF MACHINES."
-echo "    YOU PROBABLY SHOULD NOT BE RUNNING THIS COMMAND."
-echo "########################################################"
-echo
-read -rp "Are you sure you want to do this? [yN] " answer
-
-if [ "$answer" != "y" ]; then
-    echo "Wise choice."
-    exit 1
-fi
-
-HASH=$(mkpasswd -m sha-512 -R 65536 | tr -d '\n')
-
-if [ -z "$HASH" ]; then
-    echo "Didn't get a password?"
-    exit 1
-fi
-
-echo -n "$HASH" > $ROOT_PW_FILE
-chown puppet:puppet $ROOT_PW_FILE
-chmod 600 $ROOT_PW_FILE
+#!/usr/bin/env python3
+import crypt
+import subprocess
+import sys
+from getpass import getpass
+from hmac import compare_digest
+
+
+WARNING_MESSAGE = """\
+################################################################################
+This will generate a new password hash, but not update it across machines
+
+To do that, you will need to put this new password hash into the puppet
+private hiera file. Do NOT do this without input from others!
+################################################################################
+"""
+
+
+def main():
+    print(WARNING_MESSAGE)
+
+    password = getpass('New password: ')
+    password_confirm = getpass('New password confirmation: ')
+
+    if password != password_confirm:
+        print('Password and confirmation do not match, please try again')
+        sys.exit(1)
+
+    hashed_password = subprocess.check_output(
+        ('mkpasswd', '-m', 'sha-512', '-R', '65536', password),
```

Comment:
Good idea, I did briefly think about this but forgot about it when making this version. I did dismiss it at the time since it would be very short-lived, but I definitely agree stdin would be much better for this, so I'll change that.

Thanks!